src/java.base/windows/native/libjava/ProcessImpl_md.c

```diff
@@ -467,9 +467,8 @@ Java_java_lang_ProcessImpl_terminateProcess(JNIEnv *env, jclass ignored, jlong h
 JNIEXPORT jboolean JNICALL
 Java_java_lang_ProcessImpl_isProcessAlive(JNIEnv *env, jclass ignored, jlong handle)
 {
-    DWORD dwExitStatus;
-    GetExitCodeProcess((HANDLE) handle, &dwExitStatus);
-    return dwExitStatus == STILL_ACTIVE;
+    return WaitForSingleObject((HANDLE) handle, 0) /* don't wait */
+                       == WAIT_TIMEOUT;
```

Comment:
Would this be better as `isProcessAlive(handle)`?

Reply:
I don't follow. Could you explain?

Reply:
The `WaitForSingleObject(handle, 0)` seems like an indirect way to determine if the process is alive.
Whereas `isProcessAlive(handle)` seems to directly answer the question.

Reply:
sorry, I still don't get it. Are you telling me that I should use isProcessAlive to implement isProcessAlive? Because it doesn't look like a WinAPI function name,

Reply:
I take it back.
It was just wishful thinking that Windows had the obvious API.
And I've intermingled Java methods for native methods.